src/error_functions.py

```diff
@@ -1,22 +1,41 @@
 import numpy as np
 from Function import Function
+from losses import losses
 
 
 def mean_euclidean_error(predicted, target):
     """
     Computes the Mean Euclidean Error between
-    the targ vector and the output pred by the net
+    the targ vector and the output pred by the net over all patterns
 
+    :type predicted: object
     :param predicted: ndarray of shape (n, m) – Predictions for the n examples
     :param target: ndarray of shape (n, m) – Ground truth w_vals for each of n examples
     :return: loss in term of mee (Mean Euclidean Error)
     """
     if predicted.shape != target.shape:
-        raise Exception(f"Mismatching shapes in MSE: predictions shape: "
-                        f"{predicted.shape} - targets shape {target.shape}")
+        raise AttributeError(f"Mismatching shapes in MEE: predictions shape: "
+                             f"{predicted.shape} - targets shape {target.shape}")
     return np.linalg.norm(predicted - target) / target.shape[0]
```

Comment:
```suggestion
    n_patterns = predicted.shape[0]
    return np.linalg.norm(predicted - target) / n_patterns
```
Che ne dici? la shape dei target e predicted dovrebbe essere la stessa. In questo modo usiamo variabili simili per mee e mse